### PR TITLE
Allow GHC 9.10

### PR DESCRIPTION
**Pull request by xsebek**
* add GHC 9.10 to tested
* regenerate CI
* allow base 4.20

### CODE REVIEW

**Comment by xsebek** [comment]
Hi @byorgey, I tried GHC 9.10 with Swarm and noticed this package's base version range needs to be relaxed.

I am surprised this tests GHC 8.0.2 from 2017; dropping it would probably not impact anyone.

File: palette.cabal [hunk 1]
```diff
@@ -28,7 +28,7 @@ Library
                        Data.Colour.Palette.Types
                        Data.Colour.Palette.RandomColor
 
-  Build-depends:       base >= 4.2 && < 4.20,
+  Build-depends:       base >= 4.2 && < 4.21,
```
Comment (xsebek) [inline palette.cabal:31]: It might be better to allow `<5` since it is very stable. 🤔

**Comment by byorgey** [comment]
> I am surprised this tests GHC 8.0.2 from 2017; dropping it would probably not impact anyone.

I agree; I tend to just add new versions over time and never get rid of old ones until they become a problem.

File: palette.cabal [hunk 2]
```diff
@@ -28,7 +28,7 @@ Library
...
                        array >= 0.4 && < 0.6,
                        colour >= 2.3 && < 3.0,
                        containers >= 0.5 && < 0.7,
```
Comment (kostmo) [inline palette.cabal:34]: Does the `containers` bound need to be bumped?  I don't quite understand the [build failure with ghc 9.10](https://github.com/diagrams/palette/actions/runs/10653835942/job/29571614649?pr=9); in fact the following local build worked for me:
```
cabal build --with-ghc ghc-9.10.1
```


```suggestion
                       containers >= 0.5 && < 0.8,
```

Comment (byorgey) [reply palette.cabal:34]: The currently released version of `palette` on Hackage in fact has `containers < 0.8`.

**Comment by byorgey** [comment]
Closed by https://github.com/diagrams/palette/commit/933ff1de7361c83ee7ee29f127e51792cf214e0e .

**Pull request closed without merging**